tests/net/proptest.rs

```diff
@@ -0,0 +1,177 @@
+//! Proptest helpers and strategies
+//!
+//! This module houses strategies to generate (and reduce/expand) various `hbbft` and `net` related
+//! structures.
+
+use proptest::prelude::Rng;
+use proptest::strategy::{Strategy, ValueTree};
+use proptest::test_runner::{Reason, TestRunner};
+
+/// Node network topology.
+///
+/// A `NetworkTopology` describes the number of correct and faulty nodes in a network. It can also
+/// be checked, "averaged" (using the `halfway` function) and generated using `NetworkTopologyTree`.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct NetworkTopology {
+    /// Total number of nodes in network.
+    pub size: usize,
+    /// Number of faulty nodes in a network.
+    pub faulty: usize,
+}
+
+impl NetworkTopology {
+    /// Creates a new `NetworkTopology` with the supplied parameters.
+    ///
+    /// Topologies that are not sane can be created using this function.
+    pub fn new(size: usize, faulty: usize) -> Self {
+        NetworkTopology { size, faulty }
+    }
+
+    /// Checks whether the network topology satisfies the `3 * faulty + 1 <= size`
+    pub fn is_sane(&self) -> bool {
```

Comment:
Because the model guarantees correctness only with less than 1/3 of all nodes being faulty, I'd suggest using a more descriptive name instead of `is_sane` - for example, `is_fault_tolerant` or simply `is_bft`. The term "sane" is vague.

Reply:
Fair enough, I do like `is_bft` because it's short.